akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedSettings.scala

```diff
@@ -48,13 +32,17 @@ private[akka] trait EventsourcedSettings {
 
     val logOnStashing = typedConfig.getBoolean("log-stashing")
 
-    EventsourcedSettingsImpl(
-      config,
+    val journalConfig = journalConfigFor(config, journalPluginId)
+    val recoveryEventTimeout: FiniteDuration =
+      journalConfig.getDuration("recovery-event-timeout", TimeUnit.MILLISECONDS).millis
```

Comment:
Was surprised this is for both journal and snapshot store (given that they could be completely different kinds of stores etc), but I see that is how it was before so ok I guess.